### PR TITLE
Layer Spy Feature

### PR DESCRIPTION
**Pull request by bnbrown2**
Adds the N2MapSpy.js class, which allows for the creation of a layer spy. A feature that lets you compare two overlaying images

### CODE REVIEW

File: nunaliit2-js/src/main/nunaliit-es6/src/n2es6/n2mapModule/N2MapCanvas.js [hunk 1]
```diff
@@ -716,13 +718,16 @@ class N2MapCanvas {
...
 		 */
+
+		const overlayTitle = this.options.overlayTitle ? this.options.overlayTitle : 'Overlays'
 		const overlayGroup = new LayerGroup({
-			title: 'Overlays',
+			title: overlayTitle, //Overlays renames beyond just spy
 			layers: this.overlayLayers
 		});
 
+		const bgTitle = (this.options.overlayTitle) ? this.options.backgroundTitle : 'Background'
```
[the remark this answers is not in the document]
Comment (alexgao1) [reply nunaliit2-js/src/main/nunaliit-es6/src/n2es6/n2mapModule/N2MapCanvas.js:728]: Checking the wrong property (should be checking `this.options.backgroundTitle`)

**Review by alexgao1 — COMMENTED** [review]
I haven't tested this locally yet, will be doing so eventually.

**Comment by bnbrown2** [comment]
new options documentation 
"layerSpy" // if true option enables layer spy feature from N2MapSpy in the atlas module
"overlayTitle" // if set to a name will rename 'Overlays' header in layer switcher, if not set then defaults to 'Overlays'
"backgroundTitle" // if set to a name will rename 'Background' header in layer switcher, if not set then defaults to 'Background'

**Comment by billingb** [comment]
Reviewed on my local machine. Works as expected. This pull request adds the following options:
**"layerSpy"**: if true option enables layer spy feature from N2MapSpy in the atlas module. For the option to work a background raster layer and overlay raster layer must be defined.
**"overlayTitle"**: if set to a string will change 'Overlays' default header in layer switcher to string provided
**"backgroundTitle"**: if set to a string will change 'Background' header in layer switcher to string provided

**Review by billingb — APPROVED** [review]

**Pull request merged**